### ISSUE TITLE
Confusing results from Yolov8, how can I improve them?

### ISSUE THREAD

**Issue by Camerabiologist** [ultralytics/ultralytics#1785]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Dear community, 

I am working on a model that is supposed to detect a particular plant species. To do this, I collected a dataset online, which I labelled by hand (~7000 labels) and from which I recently added pseudolabels, to the classes that had comparatively few labels, using a previously trained model.

My model performance is far from where I want it to be, and some observations have made me consider that something must be wrong in my training pipeline. For instance, I trained YOLOv8l for 100 epochs yesterday on my full dataset, except it stopped training after 71 epochs, because with patience=50, the best results had been made at already the 21st epoch. As I am training a complex such as a plant, I would expect the model to require the model to need hundreds of training epochs. I wonder why this might be happening, if any of you has a good idea I would be very happy to hear it.

I suspect the problem is my dataset and not the model, because I just briefly trained COCO128, which seems to have gone fine.
![results](https://user-images.githubusercontent.com/59204167/229439190-576d02d9-4c01-472a-8f61-ac81dbbb9ac2.png).

The results from my custom dataset training, however, look like this: 
![results](https://user-images.githubusercontent.com/59204167/229462822-b34e967d-115b-4515-9fe7-9e2987638990.png)

I was using the following code to train my model: 
from ultralytics import YOLO
import os

os.environ["KMP_DUPLICATE_LIB_OK"] = "TRUE"
os.environ['CUDA_VISIBLE_DEVICES'] = '/mygpu/'
os.environ['CUDA_LAUNCH_BLOCKING'] = '1'

model = YOLO("yolov8s.yaml")  # build a new model from scratch

if __name__ == '__main__':
    model.train(data="config.yaml",
                device='0',
                epochs=100,
                batch=20)

This is the config.yaml I used:
path: C:/path/to/directory #dataset root directory
train: images/train # train images
val: images/validate # validate images
test: images/test # test images

names:
  0: class1
  1: class2
  2: class3
  3: class4
  4: class5
  5: class6
  6: class7

All of the classes have at least 2000 labels. 

If there is any other information that might be relevant to my question, please feel free to request it.

Hope someone can help me find out what's going on that could be causing the strange behaviour.

**Comment by glenn-jocher** [comment]
@Camerabiologist it's possible that your model is overfitting to your custom dataset due to having a small number of samples compared to the COCO dataset. You could try data augmentation techniques such as random crops, flips, and rotations to generate more training samples. It may also be helpful to check the learning rate and adjust the hyperparameters accordingly. You can find more information on these topics in the Ultralytics YOLOv8 documentation at docs.ultralytics.com.

**Comment by Camerabiologist** [comment]
Thanks @glenn-jocher! I'll experiment first with data augmentation and learning rate then, to see if it improves anything. Will get back with results probably tomorrow.
